src/side-by-side/components/iframe-authoring.tsx

```diff
@@ -0,0 +1,137 @@
+
+export const IframeAuthoring: React.FC<FieldProps> = props => {
+  const { onChange, formData } = props;
+  const { libraryInteractiveId, authoredState, id, navImageUrl, navImageAltText } = formData;
+  const [ iframeHeight, setIframeHeight ] = useState(300);
+  const [ authoringOpened, setAuthoringOpened ] = useState(false);
+  const interactiveWrapperClass = authoringOpened ? `${css.iframeAuthoring} ${css.open}` : css.iframeAuthoring;
+  const iframeRef = useRef<HTMLIFrameElement>(null);
+  const phoneRef = useRef<IframePhone>();
+  const iframeCurrentAuthoredState = useRef<any>();
+
+  const handleLibraryInteractiveIdChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    const newLibraryInteractiveId = event.target.value;
+    onChange({
+      libraryInteractiveId: newLibraryInteractiveId,
+      authoredState: undefined,
+      id: id || uuidv4(),
+      navImageUrl: navImageUrl || "",
+      navImageAltText: navImageAltText || ""
+    });
+  };
+
+  const handleHeaderClick = () => {
+    setAuthoringOpened(!authoringOpened);
+  };
+
+  const initInteractive = useCallback(() => {
+    const phone = phoneRef.current;
+    if (!phone) {
+      return;
+    }
+    phone.addListener("authoredState", (newAuthoredState: any) => {
+      // Save current iframe authored state.
+      iframeCurrentAuthoredState.current = newAuthoredState;
+      onChange({libraryInteractiveId,
+        authoredState: newAuthoredState,
+        id: id || uuidv4(),
+        navImageUrl: navImageUrl || "",
+        navImageAltText: navImageAltText || ""
+      });
+    });
+    phone.addListener("height", (newHeight: number) => {
+      setIframeHeight(newHeight);
+    });
+    phone.addListener("getFirebaseJWT", async (request) => {
+      const {requestId, firebase_app} = request;
+      const jwt = await getFirebaseJwt(firebase_app);
+      const response = {requestId, ...jwt};
+      phone.post("firebaseJWT", response);
+    });
+    // Nested interactives Side-By-Side interactives can't link to other top-level interactives in order to consume
+    // their data, but we want to allow a graph (or any other consumer) to request the data from the other sibling
+    // nested interactive. This response to `getInteractiveList` allows the author to select "Side-by-side Data" from
+    // the drop-down list.
+    // We can do this "blindly" without needing the id of the other interactive because the Side-by-Side runtime will
+    // be handling the data connection.
+    phone.addListener("getInteractiveList", (request)=> {
+      phone.post("interactiveList",
+        {
+          requestId: request.requestId,
+          interactives: [
+            {
+              id: LocalLinkedInteractiveId,
+              pageId: 0,
+              name: "",
+              section: "interactive_box",
+              url: "",
+              thumbnailUrl: null,
+              supportsSnapshots: false
+            }
+          ]
+        }
+      );
+    });
```

Comment:
This means that interactive within side-by-side won't be able to link to other interactives. I guess we're not hoping to create something super generic, so I guess it's fine.

Reply:
Right, but this doesn't remove any existing feature. Currently they already can't link to other interactives -- e.g. a graph in a carousel could not link to an external interactive, because this IframePhone message was simply being dropped in the IframeRuntime.  This adds the new capability of being able to listen to the sibling interactive. If we want to add a further feature down the line of being able to listen to an external interactive, we could add that later.